packages/daemon/__tests__/integration/wait-for-db-up.ts

```diff
@@ -0,0 +1,42 @@
+import mysql from 'mysql2/promise';
+import {
+  DB_USER,
+  DB_PASS,
+  DB_PORT,
+  DB_ENDPOINT,
+} from './config';
+
+const attemptConnection = async (maxAttempts: number, interval: number): Promise<void> => {
+  let attempts = 0;
+
+  while (attempts < maxAttempts) {
```

Comment:
This will never trigger, should we just make it `while (true)`?

Reply:
Why wouldn't it trigger? It might if something wrong happens with the db instance

Reply:
Because if this is false it would enter the else in the catch block below, throwing an error before checking this condition.

Also, if we exit the while loop we would make the function silently return without errors but without making a connection to the database, maybe we should remove the else condition and throw the error outside the while.
What do you think?

Reply:
I agree! Done in 95e42231d85f107e8d242872d4e36a916b688980